### DISCUSSION TITLE
How Can I Configure a Reverse Path of a Tool List?

### DISCUSSION THREAD

**Discussion by vladQll** [xournalpp/xournalpp#5054] [Q&A] [answered]
Hello I would like to know how I could configure the reverse path in a list of tools that I have, because my tablet has dials and set the right direction of travel although I also put for the left direction both always traveled according to the order of the list my query is how I could do so that the result is as follows if I turn to the right (clockwise) advance according to the order established in the list, but if I turn to the left (counterclockwise) advance in reverse order the list, this would be very useful because as my tablet has two dials on the other dial I have configured the colors but I would also like to go through it inversely as I have about 10 colors.
![image](https://github.com/xournalpp/xournalpp/assets/28522011/81e698bf-c22d-4503-b55a-585141333844)

**Comment by rolandlo** [comment, marked as answer]
You can use the same function like this:
``` lua
app.registerUi({["menu"] = "Cycle through drawing type (right)", ["callback"] = "drawingtype", ["accelerator"] = "<Alt>d", ["mode"]=1})
app.registerUi({["menu"] = "Cycle through drawing type (left)", ["callback"] = "drawingtype", ["accelerator"] = "<Alt>l", ["mode"]=-1})
``` 
and
```lua
function drawingtype(mode)
  currentDrawingtype = (currentDrawingtype + mode-1) % #drawingtypeList + 1
  ... (same as before)
```

**Edit:** Fixed typos

**Reply by vladQll** [reply]
estoy obteniendo un error 
![image](https://github.com/xournalpp/xournalpp/assets/28522011/74dd607a-6bd8-4443-8b10-3462b4103e6f)

**Reply by vladQll** [reply]
it seems it was because it was app.registerUi instead of registerUI

**Reply by rolandlo** [reply]
So does it work after fixing that typo?

**Reply by vladQll** [reply]
ese mensaje aparecio al antes de ingresar a xournal , corrigiendo si ingreso , pero justo en el instante que probe y gire hacia la derecha o izquierda me mostro un mensaje y no funciono 
![image](https://github.com/xournalpp/xournalpp/assets/28522011/b5e30a64-c9d3-4a8b-9678-772512d29240)

**Reply by rolandlo** [reply]
Ok, another typo, `drawingtypeList` is correct.

**Reply by rolandlo** [reply]
Generally speaking it would help if you posted the code as text, not as image, when you ask questions. Then I would make less typos.

**Reply by vladQll** [reply]
esta bien intentare enviar codigo , corregi el error pero obtuve un siguiente error ![image](https://github.com/xournalpp/xournalpp/assets/28522011/7d1f2a8c-3125-4698-9f3a-740966d03ab3)

este es el codigo de la linea 106:

`function drawingtype(mode) currentDrawingType = currentDrawingType + (mode-1) % #drawingtypeList + 1 app.uiAction({["action"] = "ACTION_" .. drawingtypeList[currentDrawingtype]}) print("ACTION_" .. drawingtypeList[currentDrawingtype]) end`

**Reply by rolandlo** [reply]
Another typo, correct is `currentDrawingtype`

**Reply by vladQll** [reply]
I already corrected that but I don't know why I keep getting other errors I don't know what I did wrong if it helps I will send you all the parts of the code involved.

![image](https://github.com/xournalpp/xournalpp/assets/28522011/299547c4-5904-4074-a049-b669775aaaa0)


code:

`app.registerUi({["menu"] = "Cycle through drawing type (right)", ["callback"] = "drawingtype", ["accelerator"] = "<Alt>d", ["mode"]=1});`
`app.registerUi({["menu"] = "Cycle through drawing type (left)", ["callback"] = "drawingtype", ["accelerator"] = "d", ["mode"]=-1});`

list code:
`local drawingtypeList = {"RULER", "TOOL_DRAW_ARROW", "TOOL_DRAW_DOUBLE_ARROW", "TOOL_DRAW_ELLIPSE","TOOL_DRAW_RECT", "TOOL_DRAW_SPLINE"} -- Don't include coordinate system and default tool
local currentDrawingtype = 1`

funcion code:
`function drawingtype(mode)
  currentDrawingtype = currentDrawingtype + (mode-1) % #drawingtypeList + 1
  app.uiAction({["action"] = "ACTION_" .. drawingtypeList[currentDrawingtype]})
  print("ACTION_" .. drawingtypeList[currentDrawingtype])
end`

**Reply by rolandlo** [reply]
The brackets were wrong in `currentDrawingtype = (currentDrawingtype + mode-1) % #drawingtypeList + 1`

**Reply by vladQll** [reply]
now if it worked, then I should replicate this for other functions like colors also I have a list of line styles, thank you very much.

**Reply by vladQll** [reply]
despues me ensenas los de los colores
